### PR TITLE
gameflow: add support for unobtainable pickups and kills

### LINKED ISSUE #470: Feature request: adjust the item and kill counters for unobtainable items
There is a section in the readme that explains that the end of the level counters aren't 100% faithful due to unobtainable items or inaccessible creatures.

That's nice, but I don't think many players will read it. We can do better, which is especially important in the context of the upcoming #469.

I propose to add a delta value in the gameflow for each level, `obtainable_kills` and `obtainable_items`, that defaults to 0, but for the levels with unreachable items, it's set to adjust the values to ones that casual players will understand.

So
Midas: `"unobtainable_items": 1`
Atlantis: `"unobtainable_kills": 4`

Then we could subtract these values during the stats screens.

This way level builders will never have to hardcode their item counts, and we only make an exception to broken levels.

We could also display the total amount in brackets, as in

Midas
ITEMS: 15 (16)

Note that these numbers are completely made up, I didn't look up the actual stats at the time of writing this.

### PR DESCRIPTION
**Pull request by walkawayy**
Resolves #470.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes

#### Descripton

...

### CODE REVIEW

File: src/game/gameflow.c [hunk 1]
```diff
@@ -706,6 +706,22 @@ static bool GameFlow_LoadScriptLevels(struct json_object_s *obj)
             cur->water_color.override = false;
         }
 
+        tmp_i = json_object_get_int(
+            jlvl_obj, "unobtainable_pickups", JSON_INVALID_NUMBER);
+        if (tmp_i != JSON_INVALID_NUMBER) {
+            cur->unobtainable.pickups = tmp_i;
+        } else {
+            cur->unobtainable.pickups = 0;
+        }
```
Comment (rr-) [inline src/game/gameflow.c:715]:
```c
        cur->unobtainable.pickups = json_object_get_int(
            jlvl_obj, "unobtainable_pickups", 0);
```

File: src/game/gameflow.c [hunk 2]
```diff
@@ -706,6 +706,22 @@ static bool GameFlow_LoadScriptLevels(struct json_object_s *obj)
...
+
+        tmp_i = json_object_get_int(
+            jlvl_obj, "unobtainable_kills", JSON_INVALID_NUMBER);
+        if (tmp_i != JSON_INVALID_NUMBER) {
+            cur->unobtainable.kills = tmp_i;
+        } else {
+            cur->unobtainable.kills = 0;
+        }
```
Comment (rr-) [inline src/game/gameflow.c:723]: Ditto

**Review by rr- — COMMENTED** [review]
LGTM apart from a minor cosmetic issue. Please consult with posix if we should display the physical number in brackets or not.

**Comment by walkawayy** [comment]
I'd think no because they are called unobtainable for a reason? And the user can change or remove the unobtainable entry from the gameflow to change the number if they'd like. But I'll tag @Richard-L .

**Review by rr- — APPROVED** [review]
Let's wait for posix though

**Comment by Richard-L** [comment]
Okay and so in the stats you always just want to show the corrected count, right? Or both in some way?

**Comment by walkawayy** [comment]
I think showing just the corrected count is fine. Most players aren't glitching to get  the unobtainable items in the game. And there's only 4 total, 1 of which is completely unobtainable. If they are glitching, they are probably power user enough to adjust the gameflow file if they want to show the total number of pickups including those unobtainable items.

**Comment by Richard-L** [comment]
I see. I think you're right.

Which levels have the obtainable by glitches pick-ups, and which has the entirely unobtainable one? I wonder if it would be nice to show a line at the bottom of the compass stats saying: "This level has x unobtainable pickups". Just a thought.

**Comment by walkawayy** [comment]
This list is here:
https://github.com/rr-/Tomb1Main/pull/387#issuecomment-1028246879

That sounds kinda wonky for the average player.

**Comment by Richard-L** [comment]
Hmm okay. I guess you're right it's best to keep it simple. That table you linked looks really nice though.

**Comment by walkawayy** [comment]
I removed the stat discrepancy part of the README since that no longer applies.

**Review by rr- — APPROVED** [review]

**Comment by walkawayy** [comment]
Resolved conflicts and gonna merge.

**Pull request merged**